### ISSUE TITLE
Proposal: boolean operators in plain English

### ISSUE THREAD

**Issue by laurentpayot** [elm/compiler#2031]
When I started programming in Elm I was a bit confused by the boolean operators syntax that is mixing C language legacy symbols like `&&` and `||` and English words like `not` and `xor`, as well as the Haskell-like `/=`. Not to mention that equality `==` is sometimes hard to distinguish from assignment `=` when using a ligaturized font.

Since Elm is still a work in progress I was wondering if CoffeeScript-inspired plain English operators `and`, `or`, `is` and `isnt` could be introduced in the compiler so we could turn things like
```elm
myBool = foo == 1 && bar /= 2 || not baz
```
into
```coffee
myBool = foo is 1 and bar isnt 2 or not baz
```

What are your thoughts?

**Comment by harrysarson** [comment]
Or functions:

```elm
myBool = 
    foo == 1 
        |> and (bar /= 2)
        |> or (not baz)
```

(I think `==` and `/=` are clear and commonly used enough to warrant operators)

**Comment by turboMaCk** [comment]
i think this proposal would require change in grammar of the language. This would introduce a-Z infix operators while at the moment all a-Z are prefix functions why all special characters (== =/, >=...) are infix.

**Comment by evancz** [comment]
I agree that `&&` and `||` are not ideal, but I also think that the alternatives are not ideal in other ways. Like @turboMaCk says, I think it would add a decent amount of ambiguity to the syntax of the language. That is not insurmountable on a technical level, but I personally find it confusing when using languages where certain words have special status that they do not have in other languages. I suspect others feel similarly.

A more heavy concern is that it'd be quite costly for everyone to update their code to work with this design. Anyone who has defined an `and` or `or` function, like the `Bitwise.and` and `Bitwise.or` functions, would now have code that improperly uses a language keyword. Sometimes I wish I had been more experimental with names like this when Elm was getting started (like with `Float`!) but I do not want to spend "change budget" on these kinds of things at this time.
